app/lib/features/space/dialogs/edit_space_sheet.dart

```diff
@@ -291,6 +314,16 @@ class _EditSpacePageConsumerState extends ConsumerState<EditSpacePage> {
     }
   }
 
+  // permission check
+  Future<bool> permissionCheck() async {
+    var member = await widget.space.getMyMembership();
```

Comment:
there is a potential race here, as the usage of `space` here doesn't bind to the riverpod notifier that updates it whenever the data comes down the pipe. Thus this isn't properly updating if e.g. you are on the screen and someone gives you admin priveleges - you'd have to reopen before this udpates. I don't think that is a critical problem here right now, but something to generally avoid when we have the chance ... better to use the riverpod ref.watch than handing around the object - riverpod.watch does a decent job of caching it for us ...

Reply:
Yes, that's a good take as we want to make things reactive as possible. My one approach to that for now is to just invalidate the providers to take on new data. Like this:https://github.com/acterglobal/a3/pull/707/files#r1251978913